### PR TITLE
Add hover over background colour transition in .menuLabel

### PR DESCRIPTION
**Pull request by melodicht**
To whom it may concern,

The menu buttons in the header now have a transition time of **0.5** seconds for a better user experience. However, I noticed that the transition would not appear if you immediately go onto a button that has a dropdown. 

Note that this is my first PR, so do correct me if I made any mistakes. Thank you.

below added by bradymiller:
Up For Grabs demo to demo this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Mu_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by bradymiller** [comment]
Up For Grabs demo to demo this PR is here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Mu_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
Hi @melodicht , Thanks for the PR submission. This does seem to keep things appearing smoother when using the menu.
Was hoping to get thoughts from @zbig01 and @robertdown on this.

**Comment by melodicht** [comment]
Dear @bradymiller, glad I could be of help. 

I played around a bit and managed to find a solution to the issue mentioned above. Further details can be found in the commit message. 

On a side note, I noticed that cursor: default has been used for when the navigation menu has been hovered over, whereas cursor: pointer has been used for when the administration menu has been over. 

```
/** Only apply if not an actual link **/
div.menuLabel:hover {
    cursor: default !important;
}
```

Would it not be better UX practice to use cursor: pointer for both? Anyway, just a thought.

**Comment by zbig01** [comment]
@melodicht, @bradymiller

Thank you for your input to the UX.

The addition of the  `transition: background-color 0.5s ease` to class` .menuLabel` appears to be working well. I however noticed a strange new quirk when hovering over the `Administration > Other` menu sub-item, it is now causing a vertical scroll bar to appear. This is not present when hovering over other dropdown sub-menu items. This is hardly noticeable in `ltr` languages, but becomes more noticeable in `rtl` languages. Check it out using Hebrew.

FYI when styling tabs please be aware there are two tab styles full and compact. Any change you make to the `tabs_style_full.css`  also has to be made to `tabs_style_compact.css`. Also ensure you check the result of any style change by using the three different themes, `light, manila` and any `color theme`. Check behavior in both `ltr` and `rtl` languages. Once done recheck tab behavior for all of the above using the tablet mode in chrome dev tools or on an actual tablet.

Be cautious when using float, it can cause havoc in `rtl` languages. If you need to use float on an element I would suggest using the class `oe-pull-away` which equals to` float:right` in `ltr` languages and `float:left` in `rtl `languages and `oe-pull-toward` which equals to `float:left` in `ltr `languages and `float:right` in` rtl `languages

Cheers

**Comment by melodicht** [comment]
Thanks for your kind guidance @zbig01, in regards to the vertical scroll issue, how do I change the language of the app? And does it happen because the list gets too long and extends pass the page height?

**Comment by zbig01** [comment]
@melodicht
I have not delved into why it is happening only with the above mentioned instance in the demo showing your changes, the same sub menu item does not exhibit this behavior in the main development demo.

**Comment by melodicht** [comment]
Dear @zbig01, is [this](https://imgur.com/a/K8O5K0A) the vertical scroll you are referring to? See the red arrow on the right hand side of page.

**Comment by zbig01** [comment]
Hi @melodicht

The reason for the vertical scroll bar in theme light is because of 
```
.menuEntries li .menuLabel {
    display: block;
    padding: 10px 20px;
}
```
Causes the menu item `Administration > Other` on hover to extend below the display height causing the vertical scroll bar to appear. 

This is over-ridden in the color schemes  by 
```
.menuEntries li .menuLabel {
    padding: 5px 10px!important;
}
```
resulting in a more compact dropdown menu that does not extend beyond the display height and thus no vertical scroll bar.

In short your changes to the css has not caused this behavior :)

**Comment by melodicht** [comment]
Oh I see, thanks @zbig01, for taking the time to look at it. So, just to clarify, your comment about a vertical scroll bar appearing is not due to my commit changes? Because I was about to suggest to use a 2-column dropdown for the `Administration` dropdown to make it more compact. Aside from that, as I have mentioned earlier about the `oe-pull-toward`'s `!important` glitching the further dropdown menus, do you think it is a good idea to create a class equivalent of that but without it? Thanks again for your time.

**Comment by zbig01** [comment]
Hi @melodicht

You are welcome. Please make relevant changes to `tabs_style_compact.css` as well and check that it works on all 3 themes :)

**Comment by melodicht** [comment]
Dear @zbig01, what is the difference between the full and compact CSS file? Is it that one uses the hamburger menu?

I have experimented with a smaller sized screen and the `oe-pull-toward-unimportant` class seems to be messing with the placement of the dropdown of the buttons. I was hoping to either have that class removed from the smaller screen versions or have `float:none` active on the smaller screen CSS.

**Comment by zbig01** [comment]
Hi @melodicht

Please be aware that the `interface/main/tabs/main.php` file that displays the initial page with the menu is dynamically styled using knockout.js. This has been further modified to make this  page work with smaller display devices, change the top horizontal main menu to a left sided menu, work in portrait mode and landscape mode etc. 
If you are going to edit the actual styling on this page by changing the classes you need to be sure that you have a clear understanding of the way it has been styled in order not to break what is working at present. 
It is simpler to change the declarations in the declaration-block of any css rule-set, adding new css classes is a bit more tricky.  May I suggest that you familiarize yourself with how knockout.js has been used to achieve these effects before you add/remove classes. See the comments in PR #2427 and PR #2431 to get started. I would also like to mention that the stylesheets that are used to render the styles are in `openemr/public/themes` and are complied as minified stylesheets using `npm run build`. So for any changes to stick you will need to run `npm run build` to recompile the stylesheets.

So to get you started on your journey it would be helpful to:
- Understand the difference between Frames and Tabs view - @bradymiller has made a useful [video](https://www.youtube.com/watch?v=-XtYLXFPo8g) 
- The difference between compact and full in tabs view - `Administration > Globals > Tabs Layout Theme`
- The `Responsive Vertical Menu Style for Tabs` setting in  `Administration > Globals`
- Basic understanding of how knockout.js works
- How `npm` is used to compile the stylesheets
- The three different types of stylesheets - `light, manila` and `color`
- Awareness that openEMR supports many languages some of which are right-to-left (rtl)

Once you have a good grasp of the above you will understand the reason behind testing any style change in

- [ ] - **Tabs View**

- [ ] -     Compact - Color, Light, Manila   
- [ ] -     Full  - Color, Light, Manila

- [ ] - **Frames View**  - Color, Light, Manila


This will then have to be tested for smaller displays using Responsive Vertical Menu Style. The process would then need to be repeated for rtl languages.

Cheers and happy coding :)

**Comment by robertdown** [comment]
Hi @melodicht,

Thanks for the contribution! I think the rate of `0.5s` is far, far too slow. I would say a rate of `0.1s` would be better, but even then I'd recommend an option in the settings for the user to disable the transition all together; however that would require quite a bit more work.

It's anecdotal, but as an end user I would never want an animated transition for menus when working, I want to know when I'm hovering over something as soon as I can so I can make the click faster so I can chart faster.

File: interface/themes/tabs_style_full.css [hunk 1]
```diff
@@ -360,6 +360,8 @@ z-index:10000;
     color: #333;
     padding: 15px 10px;
     font-weight: 700;
+    transition: background-color 0.5s ease;
```
Comment (robertdown) [inline interface/themes/tabs_style_full.css]: I recommend this be significantly faster, `0.1s` is where I would put it. I think `0.5s` is too slow for an EMR workflow.

**Comment by bradymiller** [comment]
Hi, Testing well, so bringing this into codebase. Thanks for the code improvements @melodicht ! -brady

**Pull request merged**